packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -150,7 +150,15 @@ export class DetachedFieldIndex {
 					});
 				}
 				this.latestRelevantRevisionToFields.delete(current);
-				this.latestRelevantRevisionToFields.set(updated, inner);
+
+				const updatedInner = this.latestRelevantRevisionToFields.get(updated);
+				if (updatedInner !== undefined) {
```

Comment:
I see, so we can run into a situation where there already is an entry in `latestRelevantRevisionToFields` for `updated`.
I think it makes sense for this to happen as part of rebasing.

Created [ADO task](https://dev.azure.com/fluidframework/internal/_workitems/edit/17086) for adding unit tests.